Researcher reports conversion page.

knowuh opened the pull request:
This PR adds code to produce a /report.html page.

On this page researchers can upload there XLSX files and have them annotated with topology data from any Sage models.

[#177674195] -- Generate topology information from a JSON structure that defines a SageModeler model.

https://www.pivotaltracker.com/story/show/177674195

dougmartin commented on README.md:
```diff
@@ -109,3 +109,28 @@ To update the icon fonts used in this project:
+### From web interface:
+1. Simply browse over to /report.html and follow the instructions
+#### Local development mode:
+6. Run a local webserver `npm run start`
```
I feel like I'm missing some steps here :)

dougmartin commented on src/code/reporting/components/app.tsx:
```diff
@@ -0,0 +1,142 @@
+
+  public render() {
+    const {running, status, CSVs} = this.state;
+    const onChangeFile = (e) => this.handleFileChange(e);
```
You could also do `this.handleFileChange = this.handleFileChange.bind(this)` in a constructor.

dougmartin commented on src/code/reporting/process-sage.ts:
```diff
@@ -0,0 +1,180 @@
+      return;
+    }
+    if (sageData.topology) {
+      topology = comp.componentStorage.savedGameState.topology;
```
this could be shortened to `topology = sageData.topology;`

dougmartin reviewed (approved):
Looks good :+1:

knowuh commented:
@dougmartin would you mind giving this another glance.  

The two commits of interest would be:  2714134   and e739a21

2714134 addresses some of your feedback
e739a21 is a change worth looking at, it basically ignores the previous topology and always recomputes.  This was based on a conversation with Dan.

dougmartin commented:
@knowuh Changes look good :+1:
